### PR TITLE
feat: application profiling

### PR DESCRIPTION
**Pull request by MuckT**
### Description

- Add documentation on how to profile release builds using Android Profiler.
- Update `Android/build.gradle` to handle optional profiling params to conditionally set profileable to true || false in `AndroidManifest.xml`
- Update `scripts/run_app.sh` to handle optional `-t` flag to set android builds as profileable.

### Other changes

- Add [react-devtools](https://www.npmjs.com/package/react-devtools) back as a dev dependency as it is now working.

### Tested

Tested locally on Android 10.

#### Devs
- Attempt to run Android Profiler using the steps outlined in the PR.
- Attempt to run react-devtools either from Flipper as or as a standalone service.
- `yarn dev:android -r -t` - will make an Alfajores Release that is profileable.
- `yarn dev:android -r -t -e mainnet` will make a Mainnet Release that is profileable.

#### QA 
- N/A

### Related issues

N/A

### Backwards compatibility

Yes

### CODE REVIEW

File: WALLET.md [hunk 1]
```diff
@@ -438,12 +439,18 @@ If you're using an Android simulator and the device / app is not showing up,
 navigate to settings (gear icon in the bottom left) and ensure the Android SDK
 location points to the same location as the $ANDROID_HOME environment variable.
 
-### App Profiling
+### App Profiling with react-devtools
 
-Run `yarn run react-devtools`. It should automatically connect to the running app, and includes a profiler (second tab). Start recording with the profiler, use the app, and then stop recording.
+From Flipper select React DevTools Plugin then while the app is running locally, or run `yarn run react-devtools` in the wallet root folder. It should automatically connect to the running app and includes a profiler (second tab). Start recording with the profiler, use the app and then stop recording. If running from the terminal, Flipper cannot be run at the same time.
```
Comment (jacob-israel-turner) [inline WALLET.md:444]:
```suggestion
From Flipper select React DevTools Plugin while the app is running locally, or run `yarn run react-devtools` in the wallet root folder. It should automatically connect to the running app and includes a profiler (second tab). Start recording with the profiler, use the app and then stop recording. If running from the terminal, Flipper cannot be run at the same time.
```

File: WALLET.md [hunk 2]
```diff
@@ -438,12 +439,18 @@ If you're using an Android simulator and the device / app is not showing up,
...
 
 The flame graph provides a view of each component and sub-component. The width is proportional to how long it took to load. If it is grey, it was not re-rendered at that 'commit' or DOM change. Details on the react native profiler are [here][rn profiler]. The biggest thing to look for are large number of renders when no state has changed. Reducing renders can be done via pure components in React or overloading the should component update method [example here][rn optimize example].
 
+### App Profiling with Android Profiler
+
+The [Android Profiler (standalone)][AndroidProfilerStandalone] is useful for viewing memory, CPU, and energy consumption. Run the profiler either from Android Studio or following the standalone instructions.
+
+Release mode is preferred for profiling as memory usage can be significantly higher in development builds. To create a  local mainnet release build for profiling run the app with `yarn dev:android -e mainnet -r -t`. Once both the app and profiler are launched attach new session selecting your device and debuggable process.
```
Comment (jacob-israel-turner) [inline WALLET.md:452]: It might be nice to specify what each flag means?

Comment (jacob-israel-turner) [reply WALLET.md:452]: The last sentence is a little confusing too. Maybe something like this? Not really sure what you're trying to say here though.

```suggestion
Release mode is preferred for profiling as memory usage can be significantly higher in development builds. To create a  local mainnet release build for profiling run the app with `yarn dev:android -e mainnet -r -t`. Once both the app and profiler are launched, attach a new session by selecting your device and a debuggable process.
```

**Review by jacob-israel-turner — APPROVED** [review]
Couple of small suggestions for the README, but looks great! 🎉

File: WALLET.md (hunk 1 above)
Comment (MuckT) [reply WALLET.md]: Changed in 878e5be8efbc6a25d4311cbb883665e9c096b835!

File: WALLET.md (hunk 2 above)
Comment (MuckT) [reply WALLET.md]: Added the descriptions of the flags and reworked the last sentence in 878e5be8efbc6a25d4311cbb883665e9c096b835!

File: scripts/run_app.sh [hunk 3]
```diff
@@ -15,13 +15,15 @@ PLATFORM=""
 ENV_NAME="alfajoresdev"
 RELEASE=false
 SIMULATOR=""
+PROFILE=false
```
Comment (jeanregisser) [inline scripts/run_app.sh:18]: nit: to clear any confusion
```suggestion
PROFILING_ENABLED=false
```

**Review by jeanregisser — APPROVED** [review]
Awesome 🚀

File: scripts/run_app.sh (hunk 3 above)
Comment (MuckT) [reply scripts/run_app.sh]: Changed in 506f0d2d2df5fb282a352c3066fe9315b0a35f35!

**Pull request merged**